deepchem/splits/splitters.py

```diff
@@ -1309,6 +1309,12 @@ def _generate_scaffold(smiles: str, include_chirality: bool = False) -> str:
         raise ImportError("This function requires RDKit to be installed.")
 
     mol = Chem.MolFromSmiles(smiles)
+    if mol is None:
+        logger.info(
+            'Not generating scaffold for smiles %s - invalid smiles string' %
+            smiles)
+        return None
```

Comment:
What happened previously when `mol` was None? We should make sure not to break existing functionality

Reply:
I throws `ValueError` and the code does not execute.

Reply:
Can you add this to the docstring to explain the difference?

Reply:
added docs in 2f35436